crypto/ct/ct_locl.h

```diff
@@ -0,0 +1,266 @@
+ */
+int SCT_get_log_entry_type(SCT *sct, log_entry_type_t *entry_type);
+
+/*
+ * Get the pointer/length of logid from the SCT and the value of the pointer
+ * to a pointer and the pointer to the length to this.
+ * The SCT retains ownership of this pointer.
+ * Returns 1 on success.
+ */
+int SCT_get0_logid(const SCT *sct, unsigned char **logid, size_t *logidlen);
+
+/*
+ * Get the timesamp of the SCT and set value of pointer to this.
+ * Returns 1 on success.
+ */
+int SCT_get_timestamp(const SCT *sct, uint64_t *timestamp);
+
+/*
+ * Return the nid for the signature used by the SCT.
+ * Currently NID_sha256WithRSAEncryption or NID_ecdsa_with_SHA256.
+ * Return -1 on failure.
+ */
+int SCT_get_signature_nid(const SCT *sct);
+
+/*
+ * Get the pointer/length of extensions from the SCT and the value of the pointer
+ * to a pointer and the pointer to the length to this.
+ * The SCT retains ownership of this pointer.
+ * Returns 1 on success.
+ */
+int SCT_get0_extensions(const SCT *sct, unsigned char **ext, size_t *extlen);
+
+/*
+ * Get the pointer/length of signature from the SCT and the value of the pointer
+ * to a pointer and the pointer to the length to this.
+ * The SCT retains ownership of this pointer.
+ * Returns 1 on success.
+ */
+int SCT_get0_signature(const SCT *sct, unsigned char **sig, size_t *siglen);
+
+/* Check SCT is valid, return 1 if valid */
+int sct_check_format(const SCT *sct);
+
+/* Free the internal datastructures of the SCT, but not the SCT itself */
+void sct_free_internal(SCT *sct);
+
+# endif
+
+/* BEGIN ERROR CODES */
+/*
+ * The following lines are auto generated by the script mkerr.pl. Any changes
+ * made after this point may be overwritten when the script is next run.
+ */
+void ERR_load_CT_strings(void);
+
+/* Error codes for the CT functions. */
+
+/* Function codes. */
+# define CT_F_SCT_CHECK_FORMAT                            112
+# define CT_F_SCT_GET0_EXTENSIONS                         100
+# define CT_F_SCT_GET0_LOGID                              101
+# define CT_F_SCT_GET0_SIGNATURE                          102
+# define CT_F_SCT_GET_LOG_ENTRY_TYPE                      103
+# define CT_F_SCT_GET_SIGNATURE_NID                       104
+# define CT_F_SCT_GET_TIMESTAMP                           105
+# define CT_F_SCT_GET_VERSION                             106
+# define CT_F_SCT_KEY_DUP                                 107
+# define CT_F_SCT_NEW                                     108
+# define CT_F_SCT_SET0_EXTENSIONS                         113
+# define CT_F_SCT_SET0_LOGID                              109
+# define CT_F_SCT_SET0_SIGNATURE                          114
+# define CT_F_SCT_SET_LOG_ENTRY_TYPE                      110
+# define CT_F_SCT_SET_SIGNATURE_NID                       115
+# define CT_F_SCT_SET_TIMESTAMP                           116
+# define CT_F_SCT_SET_VERSION                             111
+
+/* Reason codes. */
+# define CT_R_ILLEGAL_CURVE                               100
```

Comment:
Some of those codes aren't used now, would be nice to clean this up and bring them back as they're needed.